crates/solver-types/src/standards/eip7683.rs

```diff
@@ -8,6 +8,10 @@ use alloy_primitives::U256;
 use serde::{Deserialize, Serialize};
 use std::str::FromStr;
 
+/// Type alias for EIP-712 data extraction result
+type Eip712ExtractionResult<'a> =
+	Result<(&'a serde_json::Map<String, serde_json::Value>, &'a str), Box<dyn std::error::Error>>;
+
```

Comment:
Do we have an eip712.rs file in solver-types that we can put this so its shared between here and api.rs?